packages/component-catalog-spa/src/App.jsx

```diff
@@ -1,41 +1,113 @@
 // Styles
-import './App.scss';
-import { useEffect, useRef } from 'react';
+import "./App.scss";
+import { useEffect, useRef, useState } from "react";
 // Components
-import '@one-platform/opc-footer/dist/opc-footer';
-import { footer } from './assets/footer';
-import { Breadcrumb, BreadcrumbItem } from '@patternfly/react-core';
+import "@one-platform/opc-footer/dist/opc-footer";
+import { footer } from "./assets/footer";
+import {
+  Breadcrumb,
+  BreadcrumbItem,
+  Page,
+  PageSection,
+  Nav,
+  NavList,
+  NavItem,
+  PageSidebar,
+} from "@patternfly/react-core";
 // Pages
-import Main from './pages/Main';
+import Main from "./pages/Main";
+// Routes
+import { Routes, Route, Link } from "react-router-dom";
+import Description from "./Components/Description";
 
-function App() {
+const Home = () => (
+<>
+<main>
+  <h3>Components Catalog</h3>
+  <p>
+    A Unified interface to access components from Chapeaux, Patternfly
+    elements and One Platform components, which allows in increase of
+    collaboration and helps to improve component quality and development & 
+    delivery speed.
+  </p>
+  <Main />
+</main>
+</>);
+
+const breadCrumb = 
+<Breadcrumb>
+  <BreadcrumbItem to="/">Home</BreadcrumbItem>
+  <BreadcrumbItem to="/components-catalogue" isActive>
+    Component Catalog
+  </BreadcrumbItem>
+</Breadcrumb>;
+
+const App = () => {
   const footerRef = useRef();
-  useEffect( () => {
+  useEffect(() => {
     if (footerRef.current) {
       footerRef.current.opcLinkCategories = footer;
     }
-  }, [] );
+  }, []);
+
+  const [navBarItems, setNavbarItems] = useState([]);
+  useEffect(() => {
+    const repositories = [
```

Comment:
I would suggest keep this outside the component as its a constant

Reply:
This has been purposely kept inside as I don't want to add it in the dependency array for the useEffect

Reply:
hey @akhilmhdh I have tried it and it gives a warning
```
rc/App.jsx
  Line 31:9:  The 'repositories' array makes the dependencies of useEffect Hook (at line 65) change on every render. Move it inside the useEffect callback. Alternatively, wrap the initialization of 'repositories' in its own useMemo() Hook  react-hooks/exhaustive-deps
  ```

Reply:
Outside the whole component 😅  not outside useEffect

Reply:
Understood